file_handler.cc

```diff
@@ -17,26 +16,21 @@ std::unordered_map<std::string,std::string> RequestHandler::content_mappings =
 	{ "png", "image/png" }
 };
 
-std::string RequestHandler::handle_echo(size_t bytes_transferred, char* data)
-{
-	//append headers setting response and content type, and echo back in body
-	std::string response_data = "";
-	response_data += "HTTP/1.1 200 OK\nContent-Type: text/plain\n";
-	response_data += "Content-Length: ";
-	response_data += std::to_string(bytes_transferred);
-	response_data += "\n\n";
-	response_data += data;
-	return response_data;
-}
+// Constructor to have directory
+FileHandler::FileHandler(std::string directory) : directory_(directory) {}
 
-std::string RequestHandler::handle_file_server(std::string file_path)
-{
+std::string FileHandler::GenerateResponse(const HttpParser& headerInfo, const std::string& requestData) const
+{	
 	std::string response_data = "";
 
+	std::string full_path = headerInfo.get_path();
+	std::size_t second_slash_pos = full_path.find("/", 1);
+	std::string file_path = directory_ + full_path.substr(second_slash_pos + 1);
+
```

Comment:
Assumes that request handler type is always the first directory and that the path starts one / deep. /<type>/<file>. I think it's fine. Just note that it's here if we want to change it.